src/v/archival/ntp_archiver_service.cc

```diff
@@ -369,6 +355,18 @@ ss::future<> ntp_archiver::upload_until_abort() {
         }
         vlog(_rtclog.debug, "upload loop synced in term {}", _start_term);
```

Comment:
Do we need to check here once again the `may_begin_uploads()` condition? we just yielded and the sync might have advanced the term, right?

Nothing bad should happen but it should make debugging in the future easier if we log "upload loop exited because term changed" instead of "upload loop synced in term <<current_term-1>>" and then enabling the merger, scrubber and exiting after starting `upload_until_term_change`